File: DSAlgorithms/patterns/2_pointers/2_remove_duplicates.cpp
#include <iostream>
#include <bits/stdc++.h>
class remove_duplicates 
{
    public: 
        static int remove(std::vector<int>& arr)
        {
            int next_non_duplicate = 1;
            for(int i = 1; i < arr.size(); i++)
            {
                if(arr[next_non_duplicate - 1] != arr[i])
                {
                    arr[next_non_duplicate] = arr[i];
                    next_non_duplicate++;
                }
            }
            return next_non_duplicate;
        }
};
int main(int argc, char* argv[])
{
    std::vector<int> arr = {2, 3, 3, 3, 6, 9 , 9};
    std::cout<<"Array new length: "<<remove_duplicates::remove(arr)<<'\n';
    arr = {2, 2, 2, 11};
    std::cout<<"Array new length: "<<remove_duplicates::remove(arr)<<'\n';
}

File: dsa_algo/patterns/2_pointers/2_remove_duplicates.ts
const remove_duplicates = (arr: number[]) =>
{
    let next_duplicate = 1;
    let i = 1;
    while (i < arr.length)
    {
        if(arr[next_duplicate - 1] !== arr[i])
        {
            arr[next_duplicate] = arr[i];
            next_duplicate ++;
        }
        i++;
    }
    return next_duplicate;
}
console.log(remove_duplicates([2, 3, 3, 3, 6, 9, 9]));
console.log(remove_duplicates([2, 2, 2, 11]));
